hypertrace-trace-enricher/hypertrace-trace-enricher-impl/src/main/java/org/hypertrace/traceenricher/enrichment/enrichers/ApiBoundaryTypeAttributeEnricher.java

```diff
@@ -83,11 +83,12 @@ public void enrichEvent(StructuredTrace trace, Event event) {
       /*
       Determines if this is entry span is an entry to  api
       1. If the event is an ENTRY span, and
-      2. If the direct parent span is either an EXIT Span or Null, then this is the entry point.
+      2. If the direct parent span is either an EXIT Span or Null, or direct parent span service name differs from current span service name.
       */
 
       Event parentEvent = graph.getParentEvent(event);
-      if (!EnrichedSpanUtils.isEntrySpan(parentEvent)) {
+      if (!EnrichedSpanUtils.isEntrySpan(parentEvent)
```

Comment:
Here's a hypothetical: in scenarios where an envoy proxy is injected, usually there will be 2 entry spans back to back and for exits, 2 exit spans back to back. What happens then when the proxy labels the service as "test-service" and the in-app agent labels it with "test-service-foo". Technically, all these spans are in one service, given that a pod is considered as the service but now there will be 2 API boundaries instead of just one.

Reply:
Thanks @tim-mwangi. Is it not possible to fix this ? This change mainly because of multiple gateway spans like Kong, Istio etc which we don't have any control. @davexroth  what should we do ?

Reply:
Do we have any attributes from Kong, Istio that indicate that span is proxySpan?

Reply:
As the boundary between two services (which will have different service labels respectively) are based on (entry, exit) pair, is it safe to assume that two consecutive (entry, entry) or (exit, exit) pairs showing different service labels can possible due to proxy?

If, yes, we can consider below logic for determing entry/exit proxySpan if `tracer.type` attribute (or some other attribute indicating proxySpan) is not available:

```
isEntryProxySpan(currentSpan):
    isEntrySpan(currentSpan AND (foreach child of (currentSpan) -> isEntrySpan(child.span) && child.service() != currentSpan.service()) 

isExitProxySpan():
    isExistSpan() AND (isParentSpanExitSpan() && parent.serviceName != this.span.serviceName())
```

 Should we decorate at proxySpan level or inner service span level for entry API boundary?
  - Currently, on exit span, we are decorating at the proxy span level. So, for entry span, we can do the same, decorating at proxy span level.

Reply:
Discussed offline. I withdraw my concerns. I'm ok with multiple API boundaries if the service name changes.

Reply:
@kotharironak  Had offline discussion with Dave and Tim, they have agreed with this change.